nipype/algorithms/confounds.py

```diff
@@ -513,12 +509,15 @@ def _list_outputs(self):
             outputs['detrended_file'] = op.abspath(self.inputs.detrended_file)
         return outputs
 
-def regress_poly(degree, data):
+def regress_poly(degree, data, remove_mean=False, axis=-1):
     ''' returns data with degree polynomial regressed out.
-    The last dimension (i.e. data.shape[-1]) should be time.
+    Be default it is calculated along the last axis (usu. time)
     '''
     datashape = data.shape
-    timepoints = datashape[-1]
+    timepoints = datashape[axis]
+
+    if remove_mean: # i.e. regress_poly degree 0, which the following does not do
```

Comment:
@shoshber - my conditional is incorrect, it should be the other way around. 

`remove_mean`  should imply "please remove the mean signal!"